bench/pbrt-v4/optimized/aggregates.cpp.ll

```diff
@@ -6178,12 +6181,11 @@ for.end163:                                       ; preds = %for.body153
   store float %.sroa.speculated.i10.i108, ptr %agg.tmp169330.sroa.7.0.agg.tmp1.i.sroa_idx, align 8
   %agg.tmp169330.sroa.8.0.agg.tmp1.i.sroa_idx = getelementptr inbounds nuw i8, ptr %agg.tmp1.i, i64 28
   store i32 %minCostSplitBucket.1, ptr %agg.tmp169330.sroa.8.0.agg.tmp1.i.sroa_idx, align 4
-  %cmp1720.i.i = icmp eq i32 %start, %end
+  %cmp1720.i.i = icmp eq i64 %add.ptr.i328.idx, %.idx
```

Comment:
It should be folded into `icmp eq i32 %start, %end`. It is weird that it doesn't happen.

Reply:
We should further relax one-use constraint like https://github.com/llvm/llvm-project/pull/76384